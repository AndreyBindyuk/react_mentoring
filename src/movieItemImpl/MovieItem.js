import React from "react";

class MovieItem extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div key={this.props.movie.id} className="movie-item-container">
      <figure class="image"><img alt="#" src={this.props.movie.poster_src}/></figure>
      <div class="movie-title"><span>{this.props.movie.title}</span></div>
      <div className="row">
      <span>{this.props.movie.genre}</span>
      <span style={{color:"#00bbec"}}>{this.props.movie.date}</span>
      </div>
      </div>
    );
  }
}

export default MovieItem;
