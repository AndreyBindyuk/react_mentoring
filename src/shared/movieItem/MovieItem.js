import React from "react";
import { Link } from "react-router-dom";
import "./Movie.css";

export class MovieItem extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {id, title, release_date, genres, poster_path } = this.props.movie;
    return (
      
      <div className="movie-item-container">
        <Link to = {`/movie/${id}`}>
        <figure className="image">
          <img alt="#" src={poster_path} />
        </figure>
        </Link>
        <div className="movie-title">
          <span>{title}</span>
        </div>
        <div className="row">
          <span>{genres["1"]}</span>
          <span style={{ color: "#00bbec" }}>{release_date}</span>
        </div>
      </div>
    );
  }
}

export default MovieItem;
