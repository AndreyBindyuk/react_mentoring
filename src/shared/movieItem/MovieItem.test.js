import React from "react";
import { shallow } from "enzyme";
import { MovieItem } from "./MovieItem";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});

function setup() {
  const movie = {
    title: jest.fn(),
    release_date: jest.fn(),
    genres: jest.fn(),
    poster_path: jest.fn()
  };
  const props = {
    movie: movie
  };
  const enzymeWrapper = shallow(<MovieItem {...props} />);
  return {
    props,
    enzymeWrapper
  };
}

describe("<MovieItem />", () => {
  it("should render components correctly", () => {
    const { enzymeWrapper, props } = setup();
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
