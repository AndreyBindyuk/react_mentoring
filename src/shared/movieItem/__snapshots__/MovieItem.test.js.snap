// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItem /> should render components correctly 1`] = `ShallowWrapper {}`;
